src/pset/mod.rs

```diff
@@ -248,13 +253,296 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    // Common pset blinding checks
+    fn blind_checks(
+        &self,
+        inp_txout_sec: &[Option<&TxOutSecrets>],
+    ) -> Result<
+        (
+            Vec<(u64, AssetBlindingFactor, ValueBlindingFactor)>,
+            Vec<usize>,
+        ),
+        PsetBlindError,
+    > {
+        if inp_txout_sec.len() != self.inputs.len() {
+            return Err(PsetBlindError::InputTxOutSecretLen);
+        }
+
+        let mut blind_out_indices = Vec::new();
+        for (i, out) in self.outputs.iter().enumerate() {
+            if out.blinding_key.is_none() {
+                // skip checks on non-blinding outputs
+                continue;
+            }
+            if let Some(blind_index) = self.outputs[i].blinder_index {
+                if blind_index as usize >= self.inputs.len() {
+                    return Err(PsetBlindError::BlinderIndexOutOfBounds(
+                        i,
+                        blind_index as usize,
+                    ));
+                } else if inp_txout_sec[blind_index as usize].is_none() {
+                    //nothing
+                } else {
+                    // Output has corresponding input blinders
+                    blind_out_indices.push(i);
+                }
+            }
+        }
+
+        // collect input factors
+        let inp_secrets = inp_txout_sec
+            .iter()
+            .filter(|o| o.is_some())
+            .map(|o| o.unwrap())
+            .map(|o| (o.value, o.asset_bf, o.value_bf))
+            .collect::<Vec<_>>();
+
+        if blind_out_indices.is_empty() {
+            // Atleast one output should be blinded in this function
+            return Err(PsetBlindError::AtleastOneOutputBlind);
+        }
```

Comment:
I'm a little confused here -- I think it's perfectly acceptable if no outputs get blinded, right?